### PR TITLE
Client Encryption : Adds integration with Cosmos SDK 3.15.0-preview

### PR DESCRIPTION
**Pull request by anujtoshniwal**
# Pull Request Template

## Description

Integration with latest Cosmos SDK 3.15.0-preview from Encryption package. Internals dependency is removed in the latest SDK, as a result have had to mirror some code.

## Type of change

- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by kr-santosh — APPROVED** [review]
Looks good to me.

**Review by j82w — DISMISSED** [review]

File: Microsoft.Azure.Cosmos.Encryption/src/DecryptableFeedResponse.cs [hunk 1]
```diff
@@ -49,11 +49,7 @@ public override IEnumerator<T> GetEnumerator()
 
             using (responseMessage)
             {
-                // ReadFeed can return 304 in some scenarios (for example Change Feed)
-                if (responseMessage.StatusCode != HttpStatusCode.NotModified)
-                {
-                    responseMessage.EnsureSuccessStatusCode();
-                }
+                responseMessage.EnsureSuccessStatusCode();
```
Comment (abhijitpai) [inline Microsoft.Azure.Cosmos.Encryption/src/DecryptableFeedResponse.cs:52]: Why this change?

File: Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            {
+                resource = FromStream<T>(result.ResourceStream);
+            }
+
+            return new EncryptionTransactionalBatchOperationResult<T>(resource);
+        }
+
+        internal static T FromStream<T>(Stream stream)
+        {
+            using (StreamReader sr = new StreamReader(stream))
```
Comment (abhijitpai) [inline Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs:42]: Shouldn't we use the customer serializer here?

File: Microsoft.Azure.Cosmos.Encryption/src/DecryptableFeedResponse.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos.Encryption/src/DecryptableFeedResponse.cs:52]: It's explained in the following PR https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1933.

File: Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs (hunk 2 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs:42]: Please add tests to validate the custom serializer.

File: Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    using System.Net;
+    using Newtonsoft.Json;
+
+    internal sealed class EncryptionTransactionalBatchResponse : TransactionalBatchResponse
+    {
+        private readonly List<TransactionalBatchOperationResult> results;
+        private readonly TransactionalBatchResponse response;
+        private bool isDisposed = false;
+
+        public EncryptionTransactionalBatchResponse(List<TransactionalBatchOperationResult> results, TransactionalBatchResponse response)
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs:19]: Take IReadOnlyList and make a copy so that this class doesn't carry state from it's caller.

File: Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Net;
+    using Newtonsoft.Json;
+
+    internal sealed class EncryptionTransactionalBatchResponse : TransactionalBatchResponse
+    {
+        private readonly List<TransactionalBatchOperationResult> results;
+        private readonly TransactionalBatchResponse response;
+        private bool isDisposed = false;
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs:17]: no need to set to default of T

File: Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs (hunk 2 above)
Comment (anujtoshniwal) [reply Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs:42]: Thanks for pointing out! Using custom serializer. 
Also added test to verify.

File: Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs (hunk 3 above)
Comment (anujtoshniwal) [reply Microsoft.Azure.Cosmos.Encryption/src/EncryptionTransactionalBatchResponse.cs:19]: Sure

**Review by j82w — APPROVED** [review]

**Review by abhijitpai — APPROVED** [review]
